api4/user.go

```diff
@@ -531,6 +534,58 @@ func revokeSession(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func autocompleteUsers(c *Context, w http.ResponseWriter, r *http.Request) {
+	channelId := r.URL.Query().Get("in_channel")
+	teamId := r.URL.Query().Get("in_team")
+	term := r.URL.Query().Get("name")
+
+	if c.Session.GetTeamByTeamId(teamId) == nil {
+		if !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+			return
+		}
+	}
```

Comment:
This style of permissions check should not be used in APIv4. To check access to a team you just need to check they have `model.PERMISSION_VIEW_TEAM` on the team using `app.SessionHasPermissionToTeam()`